src/registryDAO.mligo

```diff
@@ -54,34 +55,41 @@ let apply_diff_affected
  *   3.b. is a token transfer
  * where min_xtz_amount and max_xtz_amount are from storage configuration.
  *)
-let registry_DAO_proposal_check (params, extras : propose_params * contract_extra) : bool =
+let registry_DAO_proposal_check (params, extras : propose_params * contract_extra) : unit =
   let proposal_size = Bytes.size(params.proposal_metadata) in
   let frozen_scale_value = unpack_nat(find_big_map("frozen_scale_value", extras)) in
   let frozen_extra_value = unpack_nat(find_big_map("frozen_extra_value", extras)) in
   let max_proposal_size = unpack_nat(find_big_map("max_proposal_size", extras)) in
 
   let required_token_lock = frozen_scale_value * proposal_size + frozen_extra_value in
-  let has_correct_token_lock =
-    (params.frozen_token = required_token_lock) && (proposal_size < max_proposal_size) in
-
-  if has_correct_token_lock then
-    match unpack_proposal_metadata(params.proposal_metadata) with
-    | Transfer_proposal tp ->
-        let min_xtz_amount = unpack_tez(find_big_map("min_xtz_amount", extras)) in
-        let max_xtz_amount = unpack_tez(find_big_map("max_xtz_amount", extras)) in
-        let is_all_transfers_valid (is_valid, transfer_type: bool * transfer_type) =
-          match transfer_type with
-          | Token_transfer_type _tt -> is_valid
-          | Xtz_transfer_type xt ->
-                 is_valid && xt.amount <> 0mutez
-              && min_xtz_amount <= xt.amount && xt.amount <= max_xtz_amount
-        in
-        List.fold is_all_transfers_valid tp.transfers has_correct_token_lock
-    | Update_receivers_proposal _urp -> has_correct_token_lock
-    | Configuration_proposal _cp -> has_correct_token_lock
-
-  else
-    false
+  let pm = unpack_proposal_metadata(params.proposal_metadata) in
+
+  let _ : unit =
+    if (params.frozen_token <> required_token_lock) then
+      fail_proposal_check (incorrect_token_amount_err_msg)
+    else if (proposal_size >= max_proposal_size) then
+      fail_proposal_check (too_large_proposal_err_msg)
+    else
+      unit in
```

Comment:
Just like for `check_xtz_transfer` this too can be moved to a common function in order to avoid calling `fail_proposal_check` twice (as well as not repeat the code for the checks).